source/postprocess/visualization.cc

```diff
@@ -320,6 +329,16 @@ namespace aspect
           && (this->get_timestep_number() != 0))
         return std::pair<std::string,std::string>();
 
+      // up the counter of the number of the file by one, but not in
+      // the very first output step. if we run postprocessors on all
+      // iterations, only increase file number in the first nonlinear iteration
+      const bool first_output = (this->get_timestep_number() == 0)
+                                && ((this->get_parameters().initial_adaptive_refinement == 0)
+                                    || (this->get_pre_refinement_step() == 0)
+                                    || (this->get_parameters().run_postprocessors_on_initial_refinement == false));
+      const bool increase_file_number = (this->get_nonlinear_iteration() == 0) || (!this->get_parameters().run_postprocessors_on_nonlinear_iterations);
```

Comment:
maybe keep the logic here. an explicit `if` would reduce readability here, and the name of the variable makes clear what the logic means